osu.Game/Screens/Edit/Setup/Components/LabelledComponents/LabelledTextBox.cs

```diff
@@ -3,127 +3,50 @@
 
 using osu.Framework.Allocation;
 using osu.Framework.Graphics;
-using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Shapes;
 using osu.Framework.Graphics.UserInterface;
 using osu.Game.Graphics;
-using osu.Game.Graphics.Sprites;
 using osu.Game.Graphics.UserInterface;
-using osuTK.Graphics;
 
 namespace osu.Game.Screens.Edit.Setup.Components.LabelledComponents
 {
-    public class LabelledTextBox : CompositeDrawable
+    public class LabelledTextBox : LabelledComponent
     {
-        private const float label_container_width = 150;
-        private const float corner_radius = 15;
-        private const float default_height = 40;
-        private const float default_label_left_padding = 15;
-        private const float default_label_top_padding = 12;
-        private const float default_label_text_size = 16;
-
         public event TextBox.OnCommitHandler OnCommit;
 
-        public bool ReadOnly
-        {
-            get => textBox.ReadOnly;
-            set => textBox.ReadOnly = value;
-        }
+        protected new OsuTextBox Component => (OsuTextBox)base.Component;
```

Comment:
any reason for casts rather than generics? also why calling this component not textbox?